src/Slices/Notification/UI/Drawer/Drawer.tsx

```diff
@@ -60,41 +77,52 @@ export const Drawer: React.FC<Props> = ({
     };
   }, [drawerRef, isDrawerOpen, onClose]);
 
-  return <View {...{ data, onClose, isDrawerOpen, trigger, drawerRef }} />;
+  return <View {...{ response, onClose, isDrawerOpen, mutate, drawerRef }} />;
 };
 
+/**
+ * Props for the View component that extends the main Drawer props.
+ *
+ * @property {UseQueryResult<NotificationResponse, Error>} response - Query result containing notification data
+ * @property {UseMutateFunction<void, Error, UpdateNotificationParams, unknown>} mutate - Function to update notifications
+ */
 interface ViewProps extends Props {
-  data: ViewData;
-  trigger(body: Body, ids: string[]): void;
+  response: UseQueryResult<NotificationResponse, Error>;
+  mutate: UseMutateFunction<void, Error, UpdateNotificationParams, unknown>;
 }
 
-const View: React.FC<ViewProps> = ({ data, onClose, trigger, drawerRef }) => {
-  const count = RemoteData.withFallback(
-    RemoteData.mapSuccess(
-      (info) => info.data.filter((n) => !n.read).length,
-      data,
-    ),
-    undefined,
-  );
+/**
+ * Internal view component that renders the notification drawer content.
+ * Handles the display of notifications and provides actions for managing them.
+ */
+const View: React.FC<ViewProps> = ({
+  response,
+  onClose,
+  mutate,
+  drawerRef,
+}) => {
+  const count = response.isSuccess
+    ? response.data.data.filter((n) => !n.read).length
+    : 0;
 
   const getOnUpdate =
     (ids: string[]): OnUpdate =>
-    async (body) => {
-      trigger(body, ids);
+    (body) => {
```

Comment:
is mutate not async? or is the async unnecessary ?

Reply:
unnecessary